### PR TITLE
Added Java time module to object mapper

### PR DESCRIPTION
**Pull request by asifsmohammed**
Signed-off-by: Asif Sohail Mohammed <nsifmoh@amazon.com>

### Description
Fix: Found an issue with peer forwarder server serialization and deserialization of `EventMetaData` with `Instant` type timeReceived.
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — CHANGES_REQUESTED** [review]
How will this impact the `MetricsConfig`? How should the date-time fields be serialized for metrics? It may be good to use a qualified `ObjectMapper` for metrics and then create an unqualified `ObjectMapper` in a more general AppConfig. This way, metrics serialization is not coupled to Data Prepper serialization.

**Comment by asifsmohammed** [comment]
I don't think it would impact anything unless an object is an `Instant`, I might be wrong here. 
Both `DataPrepperAppConfiguration` and  `MetricsConfig` uses the same ObjectMapper with `YAMLFactory`. I could create a new one in `PeerForwarderAppConfig` so that it won't be coupled to them.
I'll test the changes and push a new revision.

**Comment by dlvenable** [comment]
> I don't think it would impact anything unless an object is an `Instant`, I might be wrong here.
> Both `DataPrepperAppConfiguration` and `MetricsConfig` uses the same ObjectMapper with `YAMLFactory`. I could create a new one in `PeerForwarderAppConfig` so that it won't be coupled to them.
> I'll test the changes and push a new revision.

I'd expect that the metrics include date values. Though, I don't know if they are Java 8 dates or the old java.util.Date values.

File: data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java [hunk 1]
```diff
@@ -53,7 +52,7 @@ public YAMLFactory yamlFactory() {
 
     @Bean
```
Comment (dlvenable) [inline data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java:53]: We should name this bean.

```
@Bean(name = "metricsObjectMapper")
public ObjectMapper objectMapper(final YAMLFactory yamlFactory) {
```

And then whatever metrics-related class is using this should use the qualified form.

```
methodName(@Qualifier("metricsObjectMapper") final ObjectMapper objectMapper)
```

I'm honestly not sure what metrics code uses this.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java (hunk 1 above)
Comment (asifsmohammed) [reply data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java:53]: This is also used in `DataPrepperAppConfiguration`, even I couldn't find where and how it's being used. I don't want to rename this to "metricsObjectMapper" as it's being used to Deserialize metricsObjectMapper as well.

 I think there is a bigger issue with how these beans are created and used by other Configuration classes.

Comment (dlvenable) [reply data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java:53]: I'm not sure this is used for metrics at all. I don't see it being used by any of these metrics beans. And it is configured for YAML. I think our metrics are output in JSON.

Comment (dlvenable) [reply data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java:53]: I suggest that we just remove the `ObjectMapper` from the metrics config entirely.

Comment (dlvenable) [reply data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java:53]: Or, let's qualify the peer-forwarder `ObjectMapper` for now. That should be the safest option. We should circle back to see how we can improve this in the future.

**Review by dlvenable — APPROVED** [review]

**Pull request merged**